editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,322 @@
+		form.setLayoutData(new GridData(GridData.FILL_BOTH));
+		form.getBody().setLayout(new GridLayout(1, false));
+
+		Composite sbody = form.getBody();
+
+		tabFolder = new CTabFolder(sbody, SWT.TOP | SWT.FLAT);
+		toolkit.adapt(tabFolder, true, true);
+		tabFolder.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+		Color selectedColor = toolkit.getColors().getColor(IFormColors.SEPARATOR);
+		tabFolder.setSelectionBackground(new Color[] { selectedColor, toolkit.getColors().getBackground() },
+				new int[] { 20 }, true);
+		tabFolder.setCursor(FormsResources.getHandCursor());
+		toolkit.paintBordersFor(tabFolder);
+
+		form.setText("Advanced Properties");
+		toolkit.decorateFormHeading(form);
+
+		form.layout();
+		tabFolder.setSelection(0);
+	}
+
+	private void createClassBrowseButton(Composite composite, Text field) {
+		Button browseBeanButton = new Button(composite, SWT.PUSH);
+		browseBeanButton.setText("...");
+		browseBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				final IProject project = CamelUtils.project();
+				String className = beanConfigUtil.handleClassBrowse(project, getDisplay().getActiveShell());
+				if (className != null) {
+					field.setText(className);
+				}
+			}
+
+		});
+	}
+
+	private void createClassNewButton(Composite composite, Text field) {
+		Button newBeanButton = new Button(composite, SWT.PUSH);
+		newBeanButton.setText("+");
+		newBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				final IProject project = CamelUtils.project();
+				String value = beanConfigUtil.handleNewClassWizard(project, getDisplay().getActiveShell());
+				if (value != null) {
+					field.setText(value);
+				}
+			}
+
+		});
+	}
+
+	private void createNoArgMethodBrowseButton(Composite composite, Text field) {
+		Button browseBeanButton = new Button(composite, SWT.PUSH);
+		browseBeanButton.setText("...");
+		browseBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				Object control = modelMap.get(CamelBean.PROP_CLASS);
+				if (control != null) {
+					final IProject project = CamelUtils.project();
+					String className = (String) control;
+					String methodName = beanConfigUtil.handleNoArgMethodBrowse(project, className,
+							getDisplay().getActiveShell());
+					if (methodName != null) {
+						field.setText(methodName);
+					}
+				}
+			}
+
+		});
+	}
+
+	private void createMethodBrowseButton(Composite composite, Text field) {
+		Button browseBeanButton = new Button(composite, SWT.PUSH);
+		browseBeanButton.setText("...");
+		browseBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				Object control = modelMap.get(CamelBean.PROP_CLASS);
+				if (control != null) {
+					final IProject project = CamelUtils.project();
```

Comment:
better to avoid to use the CamelUtils.project() as it is linking this code the workbench state.
On AbstractCamelElement we can getCamelFile().getResource().getProject()